cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -51,12 +59,61 @@ public Replicator build() {
                 throw new IllegalStateException("Source and target cannot be null");
             }
 
-            PushReplication pushReplication = new PushReplication();
-            pushReplication.source = super.source;
-            pushReplication.target = super.target;
-            pushReplication.responseInterceptors.addAll(super.responseInterceptors);
-            pushReplication.requestInterceptors.addAll(super.requestInterceptors);
-            return new BasicReplicator(pushReplication, super.id);
+            BasicPushStrategy pushStrategy = new BasicPushStrategy(super.source,
+                    super.target,
+                    super.requestInterceptors,
+                    super.responseInterceptors);
+
+            pushStrategy.changeLimitPerBatch = changeLimitPerBatch;
+            pushStrategy.batchLimitPerRun = batchLimitPerRun;
+            pushStrategy.bulkInsertSize = bulkInsertSize;
+            pushStrategy.pushAttachmentsInline = pushAttachmentsInline;
+
+            return new BasicReplicator(pushStrategy, super.id);
+        }
+
+        /**
+         * Sets the number of changes to fetch from the _changes feed per batch
+         *
+         * @param changeLimitPerBatch The number of changes to fetch from the _changes feed per batch
+         * @return This instance of {@link ReplicatorBuilder}
+         */
+        public Push changeLimitPerBatch(int changeLimitPerBatch) {
+            this.changeLimitPerBatch = changeLimitPerBatch;
+            return this;
+        }
+
+        /**
+         * Sets the number of batches to push in one replication run - which is equivalent to
+         * changeLimitPerBatch * batchLimitPerRun documents
```

Comment:
Is the "which is equivalent to changeLimitPerBatch * batchLimitPerRun documents" necessary here?  That part of the comment confused me initially.